pandas/core/arrays/string_.py

```diff
@@ -0,0 +1,270 @@
+    def name(self) -> str:
+        """
+        The alias for StringDtype is ``'string'``.
+        """
+        return "string"
+
+    @classmethod
+    def construct_from_string(cls, string: str) -> ExtensionDtype:
+        if string == "string":
+            return cls()
+        return super().construct_from_string(string)
+
+    @classmethod
+    def construct_array_type(cls) -> "Type[StringArray]":
+        return StringArray
+
+    def __repr__(self) -> str:
+        return "StringDtype"
+
+
+class StringArray(PandasArray):
+    """
+    Extension array for text data.
+
+    .. versionadded:: 1.0.0
+
+    .. warning::
+
+       StringArray is considered experimental. The implementation and
+       parts of the API may change without warning.
+
+       In particular, the NA value used may change to no longer be
+       ``numpy.nan``.
+
+    Parameters
+    ----------
+    values : array-like
+        The array of data.
+
+        .. warning::
+
+           Currently, this expects an object-dtype ndarray
+           where the elements are Python strings. This may
+           change without warning in the future.
+    copy : bool, default False
+        Whether to copy the array of data.
+
+    Attributes
+    ----------
+    None
+
+    Methods
+    -------
+    None
+
+    See Also
+    --------
+    Series.str
+        The string methods are available on Series backed by
+        a StringArray.
+
+    Examples
+    --------
+    >>> pd.array(['This is', 'some text', None, 'data.'], dtype="string")
+    <StringArray>
+    ['This is', 'some text', nan, 'data.']
+    Length: 4, dtype: string
+
+    Unlike ``object`` dtype arrays, ``StringArray`` doesn't allow non-string
+    values.
+
+    >>> pd.array(['1', 1], dtype="string")
+    Traceback (most recent call last):
+    ...
+    ValueError: Must provide strings
+    """
+
+    # undo the PandasArray hack
+    _typ = "extension"
+
+    def __init__(self, values, copy=False):
+        super().__init__(values, copy=copy)
+        self._dtype = StringDtype()
+        self._validate()
+
+    def _validate(self):
+        """Validate that we only store NA or strings."""
+        if len(self._ndarray) and not lib.is_string_array(self._ndarray, skipna=True):
+            raise ValueError("StringArray requires an object-dtype ndarray of strings.")
+        if self._ndarray.dtype != "object":
+            raise ValueError(
+                "StringArray requires an object-dtype ndarray. Got "
+                "'{}' instead.".format(self._ndarray.dtype)
+            )
+
+    @classmethod
+    def _from_sequence(cls, scalars, dtype=None, copy=False):
+        if dtype:
+            assert dtype == "string"
+        result = super()._from_sequence(scalars, dtype=object, copy=copy)
+        # convert None to np.nan
+        # TODO: it would be nice to do this in _validate / lib.is_string_array
+        # We are already doing a scan over the values there.
+        result[result.isna()] = np.nan
```

Comment:
IIRC we tend to preserve ``None`` as the missing value token if a user passes an ``object`` array (rather than replacing as ``np.nan``, so am surprised this doesn't break any tests. (note that I have always found this to be weird :->)

Reply:
Yeah. I think no tests break since the `.str.<method>` tests just preserve `np.nan` or `None` through ops. And I'm *only* allowing `np.nan` here, which gets preserved.

I really don't think we want to allow `np.nan` or `None` here. It is indeed weird that we allow both.